src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotSource.java

```diff
@@ -576,28 +572,28 @@ private List<StopLocation> checkNewStopTimeUpdatesAndFindStops(
       }
 
       // Check arrival time
-      if (stopTimeUpdate.hasArrival() && stopTimeUpdate.getArrival().hasTime()) {
+      final var arrival = addedStopTime.getArrivalTime();
```

Comment:
While lots of stuff has moved into AddedStopTime, it's not well encapsulated. Here, for example, you should create a method `hasArrivalTime`.